src/tests/stdlib/text_replace_regex_ext.ab

```diff
@@ -1,17 +1,29 @@
 import * from "std/text"
 
-// Output
-// abc[123]def
-// aeon aeons ### eons
-// #path#to#file.txt
-
 main {
-    $ re='Copyright.+Free Software Foundation'; [[ \$(sed --version 2>/dev/null) =~ \$re ]] $ failed {
-        echo "Succeeded"
-    }
     // This will fail on any system where sed does not support extended
     // regex syntax, via "-r" on GNU sed and "-E" on all other versions.
-    echo replace_regex("abc123def", "([0-9]+)", "[\1]", true)
-    echo replace_regex("aeon aeons eon eons", "\beon\b", "###", true)
-    echo replace_regex("/path/to/file.txt", "/", "#", true)
+    if replace_regex("aaa", "a+", "", true) != "" {
+        echo "One or more matching failed"
+    }
+    if replace_regex("abc", "[ab]", "c", true) != "ccc" {
+        echo "Single character bracket expression failed"
+    }
+    if replace_regex("E", "[A-Z]", "E", true) != "E" {
+        echo "Range bracket expression failed"
+    }
+    if replace_regex("abc123def", "([0-9]+)", "[\1]", true) != "abc[123]def" {
+        echo "Combination of bracket exp and one or more failed"
+    }
+    if replace_regex("aeon aeons eon eons", "\beon\b", "###", true) == "aeon aeons ### eons" {
+        echo "Word boundary is handled, but it should not because it is not POSIX standard"
+    }
+    if replace_regex("/path/to/file.txt", "/", "#", true) != "#path#to#file.txt" {
+        echo "Replacing forward slash failed"
+    }
+    if replace_regex(".tar.gz", "\.(tar\.gz|tgz)$", "", true) != "" {
+        echo "A pipe (|) should be treated as OR in POSIX Standards ERE. Actually not."
+    }
```

Comment:
I know that it makes it easier to manage errors but the reason why we write `Output` comment above the code is because then we can simply echo the result. If the result is different than the expected output, then we can see what exactly happened.

If we place if statements everywhere, then now we have to run every single function twice:
1. Say why the function failed
2. Output what the function returned

I'd stick to the framework that we're already using
```
// Output
// <output1>
// <output2>
// ...

main {
  echo fun1()
  echo fun2()
}
```